lib/p2p/NodeList.ts

```diff
@@ -43,6 +43,11 @@ class NodeList extends EventEmitter {
   private static readonly BAN_THRESHOLD = -50;
   private static readonly MAX_REPUTATION_SCORE = 100;
 
+  private static readonly PRIMARY_RANK_CONN_FAILURES_THRESHOLD = 200;
```

Comment:
Minor nit, but I like "Peer" in place of "Rank" here. As is it's not clear by the name alone that this pertains to peers/nodes, although in the context of this file/class (NodeList) that may be clear enough. I'd also be fine with just dropping "Rank" from the names entirely so that it'll be a little shorter. Also OK to keep as is if you like the current names, just thought I'd mention it.

Reply:
Fixed.